src/ProjectSystemTools/TableControl/ContentWrapper.cs

```diff
@@ -38,9 +42,12 @@ internal void OpenContextMenu()
 
             // Show context menu blocks, so we need to yield out of this method
             // for e.Handled to be noticed by WPF
-            Dispatcher.BeginInvoke(new Action(() =>
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
```

Comment:
In this case, `JTF.RunAsync` might be a better fit. `BeginInvoke` essentially queues the delegate to run on the UI thread at some point in the future, and returns immediately.